### ISSUE TITLE
[Broken Build]: Nullability issue with extension blocks

### ISSUE THREAD

**Issue by michaelmairegger** [dotnet/roslyn#81699]
### Issue Description

Building the following code fails when with CS8620 but it shouldn't. If I invoke the method with t.TestMethod("Test"!) it builds

Note: If I build the code with classic extension methods the code compiles without any warnings. The issue is ralated to the new extension block only

> warning CS8620: Argument of type 'string' cannot be used for parameter 'args' of type 'string[]' in 'void extension(Test).TestMethod(params string[] args)' due to differences in the nullability of reference types.

### Steps to Reproduce

Enable nullable on project level and treat nullable warnings as errors
```
<PropertyGroup Label="CompilerInformation">
  <Nullable>enable</Nullable>
  <WarningsAsErrors>$(WarningsAsErrors);nullable</WarningsAsErrors>
</PropertyGroup>
```

Add the following lines of code and compile

```
public class InvokeTest
{
    public void Test()
    {
        var t = new Test();
        t.TestMethod("Test");
    }
}

public class Test
{
}

public static class TestExtension
{
    extension(Test test)
    {
        public void TestMethod(params string[] args)
        {
        }
    }
}
```

### Expected Behavior

No CS8620 with new C#14 extension block as with classis extension methods there is no warning

### Actual Behavior

wrong msbuild warning
> warning CS8620: Argument of type 'string' cannot be used for parameter 'args' of type 'string[]' in 'void extension(Test).TestMethod(params string[] args)' due to differences in the nullability of reference types.

### Workaround

Add to all parameters the null-forgiving operator `t.TestMethod("Test"!, "Test2"!);`

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/roslyn/issues/80320
- https://github.com/dotnet/roslyn/issues/81205
- https://github.com/dotnet/roslyn/issues/80024
- https://github.com/dotnet/roslyn/issues/81396
- https://github.com/dotnet/roslyn/issues/81394

**Comment by michaelmairegger** [comment]
this is indeed a duplicate
